### DISCUSSION TITLE
Files are not found when using Plot_Data_Plane

### DISCUSSION THREAD

**Discussion by haiPensKi** [dtcenter/METplus#1419] [File I/O] [answered]
Hi,
I have been having issues trying to run Plot_Data_Plane on Cheyenne using a wrfout file that I have post-processed using wrf_interp. I am also having issues with NEXRAD netCDF files from GridRad that I downloaded from NCAR RDA. When I run Plot_Data_Plane, I get an error message telling me that the file could not be found despite the file being in the input directory. There is something I am doing wrong, but I cannot figure it out. I am able to run Plot_Data_Plane with sample forecast data that is available in the Metplus-4.0_sample_data directory on Cheyenne. 

/glade/p/ral/jntp/MET/METplus/METplus-4.0.0/ush/run_metplus.py -c /glade/u/home/haidenm/PlotDataPlane_netcdf.conf -c /glade/u/home/haidenm/cheyenne.jpresto.conf   is the command I use.

The current PlotDataPlane_netcdf.conf is set up for the NEXRAD data.

On Cheyenne, my data files are located in /glade/u/home/haidenm/test_data. The names of NEXRAD files begin with nexrad, and the wrfout file end with _interp. 

An error log that can looked at is /glade/u/home/haidenm/logs/metplus.log.20220207083544

**Comment by jprestop** [comment, marked as answer]
Hi @haiPensKi.

I copied over your /glade/u/home/haidenm/PlotDataPlane_netcdf.conf file to my working directory and set up my cheyenne.jpresto.conf file to point to the same INPUT_BASE as in your /glade/u/home/haidenm/cheyenne.jpresto.conf:

> INPUT_BASE = /glade/u/home/haidenm/test_data

When I ran the following:
```
run_metplus.py -c PlotDataPlane_netcdf.conf -c cheyenne.jpresto.conf
```
from my working directory, I received the following error:

> Could not find INPUT file /glade/u/home/haidenm/test_data/aug272019/hrrr.t09z.wrfprsf01.grib2 using template /glade/u/home/haidenm/test_data/aug272019/hrrr.t09z.wrfprsf01.grib2

When I looked for that file, it was not there:
```
(NPL) cheyenne2:jpresto> ls /glade/u/home/haidenm/test_data/aug272019/hrrr.t09z.wrfprsf01.grib2
ls: cannot access '/glade/u/home/haidenm/test_data/aug272019/hrrr.t09z.wrfprsf01.grib2': No such file or directory
```

I ran the find command:

> find /glade/u/home/haidenm -name hrrr.t09z.wrfprsf01.grib2

but that command did not find the file.  You said that the current PlotDataPlane_netcdf.conf was set up for the NEXRAD data, but it appears it is using HRRR data instead.  When I try to run plot_data_plane on the command line with one of the nexrad files, I get the following:
```
(NPL) cheyenne2:jpresto> plot_data_plane /glade/u/home/haidenm/test_data/nexrad_3d_v4_2_20170630T160000Z.nc ./nexrad_3d_v4_2_20170630T160000Z.ps 'name="Nradobs"; level="(*,*,*)";' -v 5
DEBUG 1: Opening data file: /glade/u/home/haidenm/test_data/nexrad_3d_v4_2_20170630T160000Z.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_None".
ERROR  : 
ERROR  : plot_data_plane -> file "/glade/u/home/haidenm/test_data/nexrad_3d_v4_2_20170630T160000Z.nc" not a valid data file
ERROR  : 
```

MET can  read a few different flavors of NetCDF files:

- its internal format which is output from tools like gen_vx_mask or pcp_combine
- files following the [Climate Forecast (CF) convention](https://cfconventions.org/latest.html)
- WRF output post-processed by the WRF-interp utility

Unfortunately, that nexrad file don't appear to fall into any of those categories.

When I looked for the error in the log file you specified, /glade/u/home/haidenm/logs/metplus.log.20220207083544, I saw the following error for a different file:

> Could not find INPUT file /glade/u/home/haidenm/test_data/wrfout_d01_2019-08-27_17:00:00_interp using template /glade/u/home/haidenm/test_data/wrfout_d01_2019-08-27_17:00:00_interp

When I looked for that file, it also was not there:
```
(NPL) cheyenne2:jpresto> (NPL) cheyenne2:jpresto> ls /glade/u/home/haidenm/test_data/wrfout_d01_2019-08-27_17:00:00_interp
ls: cannot access '/glade/u/home/haidenm/test_data/wrfout_d01_2019-08-27_17:00:00_interp': No such file or directory
```

I ran the find command:

> find /glade/u/home/haidenm -name wrfout_d01_2019-08-27_17:00:00_interp

but that command did not find the file.  I do see a "wrfout_d01_2019-08-27_17:00:00_INTRP" file in your directory, however.  If you saved off your PlotDataPlane_netcdf.conf based on this wrfout data, you could try replacing **interp** with **INTRP**  and see if the code is able to successfully run with the filename fixed.

I hope that helps.  Please let us know what other questions you have.

**Reply by haiPensKi** [reply]
Hi @jprestop ,

I apologize for any difficulties you had. I thought I had saved the Plot_Data_Plane config file for the nexrad data, but it appears it did not save.  I retried running the wrfout data with _INTRP instead of _intrp, and I am able to run things successfully. The HRRR data is located in another directory, and I am now able to run the program successfully with that data. Thank you for letting me know that the nexrad file is not one MET is able to work with. 

Thank you for all of your help. Feel free to close this discussion. If I have more questions, I will start a new one.

**Reply by jprestop** [reply]
It's no trouble at all.  I'm glad to hear that you were able to run things successfully!
